napalm_ios/ios.py

```diff
@@ -1134,129 +1136,225 @@ def bgp_time_conversion(bgp_uptime):
                 is_up = False
-
-            if bgp_version == '4':
-                address_family = 'ipv4'
-            elif bgp_version == '6':
-                address_family = 'ipv6'
-            else:
-                raise ValueError("BGP neighbor parsing failed")
-
-            cmd_remote_rid = 'show ip bgp neighbors {} | inc router ID'.format(peer_id)
-            # output: BGP version 4, remote router ID 1.1.1.1
-            remote_rid_out = self._send_command(cmd_remote_rid)
-            remote_rid = remote_rid_out.split()[-1]
-
-            bgp_neighbor_data['global']['peers'].setdefault(peer_id, {})
-            peer_dict = {}
-            peer_dict['uptime'] = self.bgp_time_conversion(up_time)
-            peer_dict['remote_as'] = int(remote_as)
-            peer_dict['description'] = u''
-            peer_dict['local_as'] = local_as
-            peer_dict['is_enabled'] = is_enabled
-            peer_dict['is_up'] = is_up
-            peer_dict['remote_id'] = py23_compat.text_type(remote_rid)
-
-            cmd_current_prefixes = 'show ip bgp neighbors {} | inc Prefixes Current'.format(peer_id)
-            # output: Prefixes Current:               0          0
-            current_prefixes_out = self._send_command(cmd_current_prefixes)
-            pattern = r'Prefixes Current:\s+(\d+)\s+(\d+).*'  # Prefixes Current:    0     0
-            match = re.search(pattern, current_prefixes_out)
-            if match:
-                sent_prefixes = int(match.group(1))
-                accepted_prefixes = int(match.group(2))
-            else:
-                sent_prefixes = accepted_prefixes = -1
-
-            cmd_filtered_prefix = 'show ip bgp neighbors {} | section Local Policy'.format(peer_id)
-            # output:
-            # Local Policy Denied Prefixes:    --------    -------
-            # prefix-list                           0          2
-            # Total:                                0          2
-            filtered_prefixes_out = self._send_command(cmd_filtered_prefix)
-            sent_prefixes = int(sent_prefixes)
-            pattern = r'Total:\s+\d+\s+(\d+).*'  # Total:     0          2
-            match = re.search(pattern, filtered_prefixes_out)
-            if match:
-                filtered_prefixes = int(match.group(1))
-                received_prefixes = filtered_prefixes + accepted_prefixes
-            else:
-                # If unable to determine filtered prefixes set received prefixes to accepted
+            # overide accepted_prefixes with neighbor data if possible (since that's newer)
+            try:
+                accepted_prefixes = int(neighbor_entry['accepted_prefixes'])
+            except (ValueError, KeyError):
+                pass
+            # try and get received prefix count, otherwise set to accepted_prefixes
+            try:
+                received_prefixes = int(neighbor_entry['received_prefixes'])
+            except (ValueError, KeyError):
```

Comment:
Can't you just use `neighbor_entry.get('received_prefixes', accepted_prefixes) here?

Although maybe we already know accepted_prefixes is an int...